core/lib/zksync_core/src/state_keeper/seal_criteria/conditional_sealer.rs

```diff
@@ -27,7 +28,14 @@ impl ConditionalSealer {
             const MOCK_BLOCK_TIMESTAMP: u128 = 0;
             const TX_COUNT: usize = 1;
 
-            let resolution = sealer.should_seal(config, MOCK_BLOCK_TIMESTAMP, TX_COUNT, data, data);
+            let resolution = sealer.should_seal(
+                config,
+                MOCK_BLOCK_TIMESTAMP,
+                TX_COUNT,
+                data,
+                data,
+                ProtocolVersionId::latest(),
```

Comment:
This is incorrect to use `latest()` here. I'd make `find_unexecutable_reason` accept version id as argument. It's called only in a `tx_sender` where it's okay to assume that the current protocol version is equal to the protocol versions of the latest sealed miniblock

Reply:
I meant not just moving `ProtocolVersionId::latest` stack up but also replacing it with "protocol versions of the latest sealed miniblock" that you can load from DB

Reply:
@koloz193